From 0c03422612724a7b94e3bc431deab998d96a0d02 Mon Sep 17 00:00:00 2001
From: Tim Gardner <tim.gardner@canonical.com>
Date: Wed, 8 Oct 2008 07:36:59 -0600
Subject: [PATCH 5/6] UBUNTU: SAUCE: Correctly blacklist Thinkpad r40e in ACPI

 OriginalAuthor: martin@objectgizmos.com
 OriginalLocation: http://launchpadlibrarian.net/18243520/ubuntu-hardy-processor_idle.diff
 Bug: #278794

Upstream (Len Brown) has not agreed to add this to his ACPI repo stating that it merely covers
the root cause. http://bugzilla.kernel.org/show_bug.cgi?id=3549#c37

Carry this as a SAUCE patch until then.

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/acpi/processor_idle.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/acpi/processor_idle.c b/drivers/acpi/processor_idle.c
index 10a2d91..a60d66b 100644
--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -104,6 +104,12 @@ static int set_max_cstate(const struct dmi_system_id *id)
 /* Actually this shouldn't be __cpuinitdata, would be better to fix the
    callers to only run once -AK */
 static struct dmi_system_id __cpuinitdata processor_power_dmi_table[] = {
+	{ set_max_cstate, "IBM ThinkPad R40e", {
+	  DMI_MATCH(DMI_BIOS_VENDOR,"IBM"),
+	  DMI_MATCH(DMI_BIOS_VERSION,"1SET")}, (void *)1},
+	{ set_max_cstate, "Medion 41700", {
+	  DMI_MATCH(DMI_BIOS_VENDOR,"Phoenix Technologies LTD"),
+	  DMI_MATCH(DMI_BIOS_VERSION,"R01-A1J")}, (void *)1},
 	{ set_max_cstate, "Clevo 5600D", {
 	  DMI_MATCH(DMI_BIOS_VENDOR,"Phoenix Technologies LTD"),
 	  DMI_MATCH(DMI_BIOS_VERSION,"SHE845M0.86C.0013.D.0302131307")},
-- 
1.6.3

